### DISCUSSION TITLE
Can't install Statamic in Docker

### DISCUSSION THREAD

**Discussion by treadnought** [statamic/cms#6049] [Troubleshooting]
Following the guide at [https://statamic.dev/installing/docker](https://statamic.dev/installing/docker), after successfully creating a fresh Laravel project I follow the general instructions at [https://statamic.dev/installing/laravel#install-statamic](https://statamic.dev/installing/laravel#install-statamic).

	treadnought@LAPTOP-OCRKH1V0:~/code/statamic$ art config:clear
	Configuration cache cleared successfully.
	treadnought@LAPTOP-OCRKH1V0:~/code/statamic$ sail composer require statamic/cms --with-dependencies
	Info from https://repo.packagist.org: #StandWithUkraine
	Using version ^3.3 for statamic/cms
	./composer.json has been updated
	Running composer update statamic/cms --with-dependencies
	Loading composer repositories with package information
	Dependency guzzlehttp/guzzle is also a root requirement. Package has not been listed as an update argument, so keeping locked at old version. Use --with-all-dependencies (-W) to include root dependencies.
	Dependency laravel/framework is also a root requirement. Package has not been listed as an update argument, so keeping locked at old version. Use --with-all-dependencies (-W) to include root dependencies.
	Updating dependencies
	Your requirements could not be resolved to an installable set of packages.
	
	  Problem 1
	    - phpunit/phpunit is locked to version 9.5.20 and an update of this package was not requested.
	    - phpunit/phpunit 9.5.20 requires ext-dom * -> it is missing from your system. Install or enable PHP's dom extension.
	  Problem 2
	    - spatie/laravel-ignition is locked to version 1.2.3 and an update of this package was not requested.
	    - spatie/laravel-ignition 1.2.3 requires ext-curl * -> it is missing from your system. Install or enable PHP's curl extension.
	  Problem 3
	    - tijsverkoyen/css-to-inline-styles 2.2.4 requires ext-dom * -> it is missing from your system. Install or enable PHP's dom extension.
	    - laravel/framework v9.12.2 requires tijsverkoyen/css-to-inline-styles ^2.2.2 -> satisfiable by tijsverkoyen/css-to-inline-styles[2.2.4].
	    - laravel/framework is locked to version v9.12.2 and an update of this package was not requested.
	
	To enable extensions, verify that they are enabled in your .ini files:
	    - /etc/php/8.1/cli/php.ini
	    - /etc/php/8.1/cli/conf.d/10-opcache.ini
	    - /etc/php/8.1/cli/conf.d/10-pdo.ini
	    - /etc/php/8.1/cli/conf.d/20-calendar.ini
	    - /etc/php/8.1/cli/conf.d/20-ctype.ini
	    - /etc/php/8.1/cli/conf.d/20-exif.ini
	    - /etc/php/8.1/cli/conf.d/20-ffi.ini
	    - /etc/php/8.1/cli/conf.d/20-fileinfo.ini
	    - /etc/php/8.1/cli/conf.d/20-ftp.ini
	    - /etc/php/8.1/cli/conf.d/20-gettext.ini
	    - /etc/php/8.1/cli/conf.d/20-iconv.ini
	    - /etc/php/8.1/cli/conf.d/20-phar.ini
	    - /etc/php/8.1/cli/conf.d/20-posix.ini
	    - /etc/php/8.1/cli/conf.d/20-readline.ini
	    - /etc/php/8.1/cli/conf.d/20-shmop.ini
	    - /etc/php/8.1/cli/conf.d/20-sockets.ini
	    - /etc/php/8.1/cli/conf.d/20-sysvmsg.ini
	    - /etc/php/8.1/cli/conf.d/20-sysvsem.ini
	    - /etc/php/8.1/cli/conf.d/20-sysvshm.ini
	    - /etc/php/8.1/cli/conf.d/20-tokenizer.ini
	You can also run `php --ini` in a terminal to see which files are used by PHP in CLI mode.
	Alternatively, you can run Composer with `--ignore-platform-req=ext-dom --ignore-platform-req=ext-curl --ignore-platform-req=ext-dom` to temporarily ignore these required extensions.
	You can also try re-running composer require with an explicit version constraint, e.g. "composer require statamic/cms:*" to figure out if any version is installable, or "composer require statamic/cms:^2.1" if you know which you need.
	
	Installation failed, reverting ./composer.json and ./composer.lock to their original content.

I assume Sail has all these extensions; what am I doing wrong?

**Comment by jasonvarga** [comment]
I also assume Sail would have all these. Very strange, I'm not sure why that would happen. I just tried it and it worked fine.

What if you try `--with-all-dependencies` (or `-W` for short) instead of `--with-dependencies`?

**Reply by treadnought** [reply]
That was the first thing I tried; no luck. However! Using Sail's docker-compose.yml (not the suggested low-overhead version) does work. Could that be the issue?
